src/lib.rs

```diff
@@ -140,9 +169,46 @@ impl HexView {
             cursor: Vec2::zero(),
             data: data.into_iter().map(|u| *u.borrow()).collect(),
             state: DisplayState::Disabled,
+            config: HexViewConfig::default()
         }
     }
 
+    /// This function allows the customization of the `HexView` output.
+    ///
+    /// For options and explanation of every possible option, see the `HexViewConfig` struct.
+    ///
+    /// #Examples
+    ///
+    /// ```
+    /// # use cursive_hexview::{HexView,HexViewConfig};
+    /// let view = HexView::new();
+    /// view.set_config(HexViewConfig {
+    ///     bytes_per_line: 8,
+    ///     ..Default::default()
+    /// });
+    ///```
+    pub fn set_config(&mut self, config: HexViewConfig) {
+        self.config = config;
+    }
+
+    /// This function allows the customization of the `HexView` output. Chainable variant
+    ///
+    /// For options and explanation of every possible option, see the `HexViewConfig` struct.
+    ///
+    /// #Examples
+    ///
+    /// ```
+    /// # use cursive_hexview::{HexView, HexViewConfig};
+    /// let view = HexView::new().config(HexViewConfig {
+    ///     bytes_per_line: 8,
+    ///     ..Default::default()
+    /// });
+    ///```
+    pub fn config(mut self, config: HexViewConfig) -> Self {
+        self.config = config;
+        self
+    }
```

Comment:
Please look at this example and adapt yours accordingly

https://github.com/hellow554/cursive_hexview/blob/4168369ea2ff121520fa1a2b789ba2c0bdef4d40/src/lib.rs#L175-L177